proposals/default_optional_arguments/proposal.txt

```diff
@@ -141,8 +142,36 @@ A use case of this behavior of the function present() could be:
       ...
     end function foo
 
+5.2. intent(out)
+
+There are several different possible behaviors for intent(out)
+variables. One option is to disallow default values on optional
+intent(out) variables. However, there are use cases where it is
+desirable to return the value of a variable if a caller requests it,
+without requiring the added logic of present() and "temporary"
+variables described above.
+
+Alternatively, we might treat intent(in) and intent(out) variables
+identically, so that the following would be valid:
+
+    real function foo(a)
+      real, intent(out), optional :: a = 0
+    end function foo
+
+The benefit is that, just like proposed for intent(in) variables,
+intent(out) variables would be usable regardless of whether the value
+is actually passed back to the caller, removing the need for present()
+and "temporary" variables. The complication is that a present
+intent(out) variable is initially undefined, meaning the logic of
+present() is still necessary in this case. Furthermore, the behavior
+suggested above where present() = .true. always for variables with a
+specified default would not be desired for intent(out) variables.
+
```

Comment:
```suggestion
A possible use case for this behaviorcould be:

    integer function open(filename, mode, iostat) result(u)
      character(*), intent(in) :: filename
      character(*), intent(in), optional :: mode
      integer, intent(out), optional :: iostat = 0
      character(3) :: mode_
      character(:),allocatable :: action_, position_, status_, access_, form_
     !some code
     !....
     open(newunit=u, file=filename, &
         action = action_, position = position_, status = status_, &
         access = access_, form = form_, &
         iostat = iostat)
end function

```

This code is actually extracted from [stlib](https://github.com/fortran-lang/stdlib/blob/1e0e460c78dec6e926b51d29fa5db889d5592f7a/src/stdlib_experimental_io.fypp#L233)

Reply:
"behaviorcould" -> "behavior could"

Reply:
I modified @jvdp1's comment to fix the spelling error and committed.